loadbalancer/zk/publisher_test.go

```diff
@@ -43,13 +44,16 @@ func TestBadFactory(t *testing.T) {
 	}
 	defer p.Stop()
 
+	// instance1 came online
+	client.AddService(path+"/instance1", "zookeeper_node_data")
+	time.Sleep(1 * time.Millisecond)
```

Comment:
We see lots of problems on Travis when we assume 1ms is enough for the scheduler to get something to happen. Are you sure this is safe long-term?